iceoryx2-cal/tests/zero_copy_connection_trait_tests.rs

```diff
@@ -324,24 +336,135 @@ mod zero_copy_connection {
+                    sut_receiver.release(sample.unwrap(), ChannelId::new(channel_id)),
+                    is_ok
+                );
+                let retrieval = sut_sender.reclaim(id).unwrap();
+                assert_that!(retrieval, is_some);
+                assert_that!(retrieval.as_ref().unwrap().offset(), eq sample_offset);
+
+                let retrieval = sut_sender.reclaim(id).unwrap();
+                assert_that!(retrieval, is_none);
+            }
+        }
+    }
+
+    #[test]
+    fn same_offset_can_be_sent_receive_and_retrieval_via_multiple_channels_in_parallel<
+        Sut: ZeroCopyConnection,
+    >() {
+        const NUMBER_OF_CHANNELS: usize = 7;
+        const ITERATIONS: usize = 5;
+        let name = generate_name();
+        let config = generate_isolated_config::<Sut>();
+
+        let sut_sender = Sut::Builder::new(&name)
+            .number_of_samples_per_segment(NUMBER_OF_SAMPLES)
+            .number_of_channels(NUMBER_OF_CHANNELS)
+            .config(&config)
+            .create_sender()
+            .unwrap();
+        let sut_receiver = Sut::Builder::new(&name)
+            .number_of_samples_per_segment(NUMBER_OF_SAMPLES)
+            .number_of_channels(NUMBER_OF_CHANNELS)
+            .config(&config)
+            .create_receiver()
+            .unwrap();
+
+        for iteration in 0..ITERATIONS {
+            let sample_offset = SAMPLE_SIZE * (iteration + 1);
+            // send out first the same sample_offset throught all channels
+            for channel_id in 0..NUMBER_OF_CHANNELS {
+                let id = ChannelId::new(channel_id);
+
+                assert_that!(
+                    sut_sender.try_send(PointerOffset::new(sample_offset), SAMPLE_SIZE, id),
+                    is_ok
+                );
+            }
+
+            let mut samples = vec![];
+            // receive the sample_offset on all channels
+            for channel_id in 0..NUMBER_OF_CHANNELS {
+                let id = ChannelId::new(channel_id);
+                let sample = sut_receiver.receive(id).unwrap();
+                assert_that!(sample, is_some);
+                assert_that!(sample.as_ref().unwrap().offset(), eq sample_offset);
+                samples.push(sample);
+            }
+
+            // release the received samples
+            for channel_id in 0..NUMBER_OF_CHANNELS {
+                let id = ChannelId::new(channel_id);
+                assert_that!(
+                    sut_receiver.release(samples[channel_id].unwrap(), id),
+                    is_ok
+                );
+            }
+
+            // reclaim them
+            for channel_id in 0..NUMBER_OF_CHANNELS {
+                let id = ChannelId::new(channel_id);
+                let retrieval = sut_sender.reclaim(id).unwrap();
+                assert_that!(retrieval, is_some);
+                assert_that!(retrieval.as_ref().unwrap().offset(), eq sample_offset);
+            }
+        }
+    }
+
```

Comment:
Can we also have a test where for instance send and receive don't work because of mismatched channel ids?